File: solutions/python/resistor-color/2/resistor_color.py
def color_code(color):
    """Calls a function that lists all colours and returns the index (value) of the band in that colour
    
    :param color: The color for the band that we want to check its value.
    :return: int - The value of that band.
    """
    
    return colors().index(color)


def colors():
    """Gives a list of all the resistor band colours
    
    :return: list - The list of resistor colours ordered by the associated values (given the index).
    """
    
    return ["black", "brown", "red", "orange", "yellow", "green", "blue", "violet", "grey", "white"]

File: solutions/php/resistor-color/1/ResistorColor.php
<?php
declare(strict_types=1);

/**
* Gets all the colours associated with resistors
* @return array<string> The list of colours.
*/
function getAllColors(): array
{
    return ["black", "brown", "red", "orange", "yellow", "green", "blue", "violet", "grey", "white"];
}

/**
* Looks up the value associated to the colour of the resistor
* @param string $color The colour that is to be consulted.
* @return int The value associated to the colour.
*/
function colorCode(string $color): int
{
    return array_search($color, getAllColors());
}
